File: Algorithms/DegreeCentrality/DegreeCentrality.py
def degree_centrality(adj_mat, index):
    count = 0
    for p in range(number_nodes):
        if adj_mat[index][p] == 1:
            count = count + 1
    return count


def print_centrality(adj_mat):
    for n in range(number_nodes):
        print(str(degree_centrality(adj_mat, n)) + "\n")
    return None


first_line = input()
first_line = first_line.split()
number_nodes = int(first_line[0])
number_edges = int(first_line[1])
adj_matrix = [[0 for x in range(number_nodes)] for y in range(number_nodes)]

for i in range(number_edges):
    my_line = input()
    my_line = my_line.split()
    my_line = [int(i) for i in (my_line)]
    adj_matrix[my_line[0]][my_line[1]] = 1
    adj_matrix[my_line[1]][my_line[0]]= 1
    i = i + 1

print_centrality(adj_matrix)

File: Algorithms/DegreeCentrality/degree centrality.cpp
#include<iostream>
#include<unordered_map>
#include<vector>
#include <string>
using namespace std;
int main() 
{
	unordered_map<string, int>graph;
	unordered_map<int, int>freq;
	int number_nodes;
	int number_edges;
	cin >> number_nodes;
	cin >> number_edges;
	for (int i = 0; i < number_edges; i++)
	{
		int u, v, w;
		cin >> u >> v >> w;
		string my_key = "";
		my_key = to_string(u) + "," + to_string(v) + "," + to_string(w);
		if (graph.count(my_key)==0)
		{
			graph[my_key] = 1;

		}
		if (freq.count(u) > 0)
		{
			freq[u]+=1;
		}
		else
			freq[u] = 1;
		if (freq.count(v) > 0)
		{
			freq[v]+=1;
		}
		else
			freq[v] = 1;
	}
	for (int i = 0; i < number_nodes; i++)
	{
		if (freq.count(i) != 0)
			cout << freq[i] << "\n";
		else
			cout << 0 << "\n";

	}
	int x;
	cin >> x;
}